classes/webservices/plagiarism_copyleaks_webhooks.php

```diff
@@ -0,0 +1,174 @@
+    $moodleuserid,
+    $identifier,
+    $scanid,
+    $status,
+    $plagiarismscore = null,
+    $aiscore = null,
+    $writingfeedbackissues = null,
+    $ischeatingdetected = false,
+    $errormessage = null
+  ) {
+    global $DB;
+
+    // Validate parameters
+    $params = self::validate_parameters(self::update_report_webhook_parameters(), array(
+      'coursemoduleid' => $coursemoduleid,
+      'moodleuserid' => $moodleuserid,
+      'identifier' => $identifier,
+      'scanid' => $scanid,
+      'status' => $status,
+      'plagiarismscore' => $plagiarismscore,
+      'aiscore' => $aiscore,
+      'writingfeedbackissues' => $writingfeedbackissues,
+      'ischeatingdetected' => $ischeatingdetected,
+      'errormessage' => $errormessage
+    ));
+
+    $result =  \plagiarism_copyleaks_submissions::update_report(
+      $params['coursemoduleid'],
+      $params['moodleuserid'],
+      $params['identifier'],
+      $params['scanid'],
+      $params['status'],
+      $params['plagiarismscore'],
+      $params['aiscore'],
+      $params['writingfeedbackissues'],
+      $params['ischeatingdetected'],
+      $params['errormessage'],
+    );
+    return array('success' => $result);
+  }
+
+  /**
+   * Describes the return value for Update_report_webhook
+   * @return external_single_structure
+   */
+  public static function update_report_webhook_returns() {
+    return new external_single_structure(
+      array(
+        'success' => new external_value(PARAM_BOOL, 'Status of the operation')
+      )
+    );
+  }
+
+
+  /**
+   * Returns description of method parameters
+   * @return external_function_parameters
+   */
+  public static function update_failed_scan_to_queued_webhook_parameters() {
+    return new external_function_parameters(
+      array(
+        'coursemoduleid' => new external_value(PARAM_TEXT, 'Course module ID'),
+        'moodleuserid' => new external_value(PARAM_TEXT, 'Moodle user ID'),
+        'identifier' => new external_value(PARAM_TEXT, 'Identifier'),
+      )
+    );
+  }
+
+  /**
+   * Update failed scan to queued webhook
+   * @param  string $coursemoduleid  Course module ID
+   * @param  string $moodleuserid    Moodle user ID
+   * @param  string $identifier      Identifier
+
+   * @return array  Warnings and success status
+   */
+
+  public static function update_failed_scan_to_queued_webhook(
+    $coursemoduleid,
+    $moodleuserid,
+    $identifier
+  ) {
+    global $DB;
+    // Validate parameters
+    $params = self::validate_parameters(self::update_failed_scan_to_queued_webhook_parameters(), array(
+      'coursemoduleid' => $coursemoduleid,
+      'moodleuserid' => $moodleuserid,
+      'identifier' => $identifier
+    ));
+
+    $fileid = $DB->get_field('plagiarism_copyleaks_files', 'id', array('cm' => $params['coursemoduleid'], 'userid' => $params['moodleuserid'], 'identifier' => $params['identifier']));
+    plagiarism_copyleaks_submissions::change_failed_scan_to_queued($fileid);
```

Comment:
Why set failed scan to queue? shouldn't we use the update_report function you created.